crates/matrix-sdk-common/src/js_tracing.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt, fmt::Write};
+
+use tracing::{
+    field::{Field, Visit},
+    Event, Level, Subscriber,
+};
+use tracing_subscriber::layer::Context;
+
+//! Utilities for `tracing` in JS environments
+
+/// An implementation of `tracing_subscriber::layer::Layer` which directs all
+/// events to the JS console
+#[derive(Debug)]
+pub struct Layer {}
+
+impl Layer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for Layer {
+     fn default() -> Self {
+         Self::new()
+     }
+ }
```

Comment:
This impl can be derived. Also I really don't like that clippy lint (am surprised it's not disabled), but here I guess it could make sense to derive `Default`.

Reply:
have updated to derive it